benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/scheduler/TestTaskScheduler.java

```diff
@@ -172,44 +189,51 @@ private synchronized void handleTestRunningState(String testID) {
       logger
           .info("handleTestRunningState for " + testID + " with state " + testRunningState.name());
 
+      // set timeout if not already set
+      if (!timeoutTasks.containsKey(testID)) {
+        Time maxRunTime = testModelDAO.getMaxRunningTime(testID);
+        TimeoutTask timeoutTask = new TimeoutTask(testID, this);
+        ScheduledFuture<?> timeoutFuture = timeoutScheduledThreadPoolExecutor.schedule(timeoutTask,
+            maxRunTime.toSecondsPart(), TimeUnit.SECONDS);
+        timeoutTasks.put(testID, timeoutFuture);
+      }
+
       switch (testRunningState) {
         case DETERMINE_EXPLORATION_STRATEGY:
-          determineExplorationStrategy(testID);
+          runningStatesHandler.determineExplorationStrategy(testID);
           break;
 
         case ADD_STORED_KNOWLEDGE:
-          addStoredKnowledge(testID);
+          runningStatesHandler.addStoredKnowledge(testID);
           break;
 
         case DETERMINE_EXECUTE_VALIDATION_SET:
-          determineExecuteInitialValidationSet(testID);
+          runningStatesHandler.determineExecuteInitialValidationSet(testID);
           break;
 
         case DETERMINE_EXECUTE_EXPERIMENTS:
-          determineAndExecuteExperiments(testID);
+          runningStatesHandler.determineAndExecuteExperiments(testID);
           break;
 
         case HANDLE_EXPERIMENT_RESULT:
-          handleExperimentResult(testID);
+          runningStatesHandler.handleExperimentResult(testID);
           break;
 
         case VALIDATE_TERMINATION_CRITERIA:
-          validateTerminationCriteria(testID);
+          runningStatesHandler.validateTerminationCriteria(testID);
           break;
 
         case DERIVE_PREDICTION_FUNCTION:
           testModelDAO.setTestRunningState(testID, VALIDATE_PREDICTION_FUNCTION);
```

Comment:
It actually happens also happens in the function in the next line. Although this wasn't changed in this PR.